### DISCUSSION TITLE
allow running from TrueNAS Scale "Launch Docker Image" button

### DISCUSSION THREAD

**Discussion by blacksteel1288** [nextcloud/all-in-one#2900] [Ideas] [answered]
I run several different docker images from the "Launch Docker Image" button on TrueNAS Scale, which avoids the need to create a Helm chart or manage upgrades through one of the other TrueNAS providers, who may or may not provide reliable updates.

It would be great to be able to run the Nextcloud AIO image in the same way, running directly from there, where when a new image is available on the latest tag, it can be restarted and download the new image and upgrade automatically, as are the other images I run.

I attempted to do this recently using the current Nextcloud AIO image, but it failed as docker is not running on TrueNAS Scale (it's k3s).  I believe there is a manual install version of Nextcloud AIO that doesn't require docker, but it comes with too much loss of functionality -- namely very manual updates where you have to compare .yml files, etc.

I also attempted to run Nextcloud AIO inside a Docker-in-Docker (dind) image that was launched using the "Launch Docker Image" button, and I was almost able to get it to work save a few networking-related issues.

Looking forward to hearing from you!  AIO is a great idea!

**Comment by szaimen** [comment, marked as answer]
Hi, due to the architecture of Truenas Scale and AIO, I don't see any way to run AIO there with just one click. However see https://github.com/nextcloud/all-in-one#can-i-run-aio-on-truenas-scale

**Reply by blacksteel1288** [reply]
Thanks, I did see that comment in the readme file.

Instead of using a VM, which might be a higher cost way, you can launch a docker image of Docker-in-Docker (https://hub.docker.com/_/docker) which gives you access to the docker socket, and then use docker-compose inside that container to run AIO, which is what I did. It can work that way, and might be faster than a VM, but needs some guidance on configuration of ports and volumes, under that kind of config.

**Reply by szaimen** [reply]
I see but since docker in docker is a disrecommended use case for docker is this not going to be implemented by us. However see https://github.com/nextcloud/all-in-one/discussions/588#discussioncomment-4127926

**Reply by blacksteel1288** [reply]
Ah, darn.  Thank you for considering.
